cmd/kubefwd/services/services.go

```diff
@@ -246,179 +242,245 @@ type FwdServiceOpts struct {
 	Context      string
 	Namespace    string
 	ListOptions  metav1.ListOptions
-	Hostfile     *txeh.Hosts
+	Hostfile     *fwdport.HostFileWithLock
 	ClientConfig *restclient.Config
 	RESTClient   *restclient.RESTClient
 	ShortName    bool
 	Remote       bool
 	IpC          byte
+	IpD          int
 	ExitOnFail   bool
 	Domain       string
 }
 
-func fwdServices(opts FwdServiceOpts) error {
+func (opts *FwdServiceOpts) StartListen(stopListenCh <-chan struct{}) {
 
-	services, err := opts.ClientSet.CoreV1().Services(opts.Namespace).List(opts.ListOptions)
+	optionsModifier := func(options *metav1.ListOptions) {
+		options.FieldSelector = fields.Everything().String()
+		options.LabelSelector = opts.ListOptions.LabelSelector
+	}
+	watchlist := cache.NewFilteredListWatchFromClient(opts.RESTClient, "services", v1.NamespaceDefault, optionsModifier)
```

Comment:
v1.NamespaceDefault cause the bug.
i forget to pass the opts.namespace args here and use default instead.